src/AlertDialog.js

```diff
@@ -88,7 +88,7 @@ class AlertDialog extends Dialog {
     let found = false;
     let index = 0;
     while (index < this.choices.length && !found) {
-      if (this.choices[index].charCodeAt(0) === event.keyCode) {
+      if (event.key.length === 1 && this.choices[index][0] === event.key) {
```

Comment:
Note that here (and everywhere else we assume `event.key.length === 1` means it's a printable character) we'll run up against [this issue](https://stackoverflow.com/questions/43877434/determine-if-key-press-resulted-in-printable-unicode-character-being-input-witho#comment75161364_44052348) with Unicode astral characters.

Reply:
Also, the `choices` comparison here is still (no change from before) case sensitive but probably shouldn't be. `toLowerCase`/`toUpperCase` causes [problems](https://stackoverflow.com/questions/26877514/is-it-better-to-compare-strings-using-tolowercase-or-touppercase-in-javascript) in various languages, so you'd have to use [`localeCompare`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) with the `sensitivity` option. (Same issue will happen with KeyboardPrefixSelectionMixin's use of toLowerCase)

Reply:
See note below.

Interesting point about case-sensitivity. The existing AlertDialog behavior is Shift-key insensitive: if you press the "O" key without Shift, it still matches "OK", because the `keyCode` corresponds to the capital letter "O" whether or not the Shift key is pressed. But the existing code fails to handle the case where the initial letter of a choice button is lowercase. If the button was labeled "ok", I'd still want "O" to match that. I can look into that after this PR lands.

Point noted about potential international issues with `toLowerCase`.

Reply:
Oh, right, I forgot that keyCode is automatically case-insensitive. Sounds good.